spec/scenarios/invoices/invoice_numbering_spec.rb

```diff
@@ -0,0 +1,182 @@
+      numbers = invoices.pluck(:number)
+
+      expect(sequential_ids).to match_array([3, 3, 3])
+      expect(organization_sequential_ids).to match_array([1, 2, 3])
+      expect(numbers).to match_array(%w[ORG-1-001-003 ORG-1-002-003 ORG-1-003-003])
+    end
+
+    # NOTE: October 19th: Switching to per_organization numbering and Bill subscription
+    travel_to(DateTime.new(2023, 10, 19, 12, 12)) do
+      organization.update!(document_numbering: 'per_organization')
+
+      Subscriptions::BillingService.call
+      perform_all_enqueued_jobs
+
+      invoices = organization.invoices.order(created_at: :desc).limit(3)
+      sequential_ids = invoices.pluck(:sequential_id)
+      organization_sequential_ids = invoices.pluck(:organization_sequential_id)
+      numbers = invoices.pluck(:number)
+
+      expect(sequential_ids).to match_array([4, 4, 4])
+      expect(organization_sequential_ids).to match_array([1, 2, 3])
+      expect(numbers).to match_array(%w[ORG-1-202310-001 ORG-1-202310-002 ORG-1-202310-003])
+    end
+
+    # NOTE: November 19th: Switching to per_customer numbering and Bill subscription
+    travel_to(DateTime.new(2023, 11, 19, 12, 12)) do
+      organization.update!(document_numbering: 'per_customer')
+
+      Subscriptions::BillingService.call
+      perform_all_enqueued_jobs
+
+      invoices = organization.invoices.order(created_at: :desc).limit(3)
+      sequential_ids = invoices.pluck(:sequential_id)
+      organization_sequential_ids = invoices.pluck(:organization_sequential_id)
+      numbers = invoices.pluck(:number)
+
+      expect(sequential_ids).to match_array([5, 5, 5])
+      expect(organization_sequential_ids).to match_array([1, 2, 3])
+      expect(numbers).to match_array(%w[ORG-1-001-005 ORG-1-002-005 ORG-1-003-005])
+    end
+
+    # NOTE: November 22: New subscription for second customer
+    time = DateTime.new(2023, 11, 22, 12, 12)
+    travel_to(time) do
+      create_subscription(
+        {
+          external_customer_id: customer_second.external_id,
+          external_id: 'new_external_id',
+          plan_code: yearly_plan.code,
+          billing_time: 'anniversary',
+          subscription_at: time.iso8601,
+        },
+      )
+
+      invoices = organization.reload.invoices.order(created_at: :desc)
+
+      expect(invoices.first.sequential_id).to eq(6)
+      expect(invoices.first.organization_sequential_id).to eq(4)
+      expect(invoices.pluck(:number))
+        .to match_array(
+          %w[
+            ORG-1-001-001
+            ORG-1-002-001
+            ORG-1-003-001
+            ORG-1-001-002
+            ORG-1-002-002
+            ORG-1-003-002
+            ORG-1-001-003
+            ORG-1-002-003
+            ORG-1-003-003
+            ORG-1-202310-001
+            ORG-1-202310-002
+            ORG-1-202310-003
+            ORG-1-001-005
+            ORG-1-002-005
+            ORG-1-003-005
+            ORG-1-002-006
+          ]
```

Comment:
Style/TrailingCommaInArguments: Put a comma after the last parameter of a multiline method call. (https://rubystyle.guide#no-trailing-params-comma)